src/esssans/i_of_q.py

```diff
@@ -70,11 +68,10 @@ def preprocess_monitor_data(
         )
         background = mask_range(monitor, mask=mask).mean()
 
-    if wavelength_bins is not None:
-        if monitor.bins is not None:
-            monitor = monitor.hist(wavelength=wavelength_bins)
-        else:
-            monitor = monitor.rebin(wavelength=wavelength_bins)
+    if monitor.bins is not None:
+        monitor = monitor.hist(wavelength=wavelength_bins)
+    else:
+        monitor = monitor.rebin(wavelength=wavelength_bins)
     if background is not None:
```

Comment:
It's an interesting exercise to decide when multiple things should go in a function, and when they should be split into separate nodes in the graph. For example, could this background be an optional part of the graph? (using an `Optional` type?)